lib/LoginStrategy.ts

```diff
@@ -251,23 +251,41 @@ export class LoginStrategy extends BaseStrategy {
   async offchainClaimsOf(did: string): Promise<OffchainClaim[]> {
     did = this.didUnification(did);
 
-    const transformClaim = (claim: OffchainClaim): OffchainClaim => {
+    const transformClaim = (
+      claim: OffchainClaim
+    ): OffchainClaim | undefined => {
       const transformedClaim = { ...claim };
-      const didFormatFields = ['iss', 'sub', 'subject', 'did'];
+      const didFormatFields = ['iss', 'sub', 'subject', 'did', 'signer'];
+
+      let isMaliciousClaim = false;
 
       Object.keys(transformedClaim).forEach((key) => {
         if (didFormatFields.includes(key)) {
-          transformedClaim[key] = this.didUnification(transformedClaim[key]);
+          try {
+            transformedClaim[key] = this.didUnification(transformedClaim[key]);
+          } catch (err) {
+            if (err instanceof Error && err.message === 'Invalid DID') {
+              isMaliciousClaim = true;
+              return;
+            }
+            throw err;
+          }
```

Comment:
Rather than catching the exception, I think it would be better if we test the case first. We can use the helper function from `ew-did-registry`
```suggestion
            const expectedEthrDID = transformedClaim[key];
            if (isValidErc1056(expectedEthrDID)) {
              transformedClaim[key] = this.didUnification(transformedClaim[key]);
            }
            else {
              isMaliciousClaim = true;
            }
            return;
```